File: Jazz.web.frame/java/jazz.common.enitiy/src/org/jazz/common/enitiy/EnitiyCache.java
package org.jazz.common.enitiy;

import java.lang.reflect.Field;
import java.util.HashMap;
import java.util.Map;

import org.jazz.common.enitiy.attribute.DBItemAttribute;
import org.jazz.common.enitiy.attribute.DBModelAttribute;

/**
 * 实体缓存子项
 * @author wjc
 *
 */
public class EnitiyCache {
	
	private Class<?> cls;
	
	private String TBName;
	
	private String[] SimpleCol;
	
	private String ItemCols;
	
	private String ListCols;
	
	private Map<String,Field> fieldMap = new HashMap<String,Field>();
					
	public Map<String,Field> getfieldMap()
	{
		return fieldMap;
	}
	
	public Field getFieldName(String Colname)
	{
		return this.fieldMap.get(Colname);
	}
	
	public String getColName(String Fieldname)
	{
		for (Map.Entry<String,Field> entry :fieldMap.entrySet()) {
		    if(entry.getValue().getName().equals(Fieldname))
		    	return entry.getKey();
		}
		return "";

	}

	public String getTBName() {
		return TBName;
	}

	public String[] getSimpleCol() {
		return SimpleCol;
	}

	public String getItemCols() {
		return ItemCols;
	}

	public String getListCols() {
		return ListCols;
	}


	public Class<?> getCls() {
		return cls;
	}

	
	public void setCls(Class<?> cls)  {
		this.cls = cls;
		this.fieldMap.clear();
		for(Field f:cls.getDeclaredFields()){
			DBItemAttribute _atr=f.getAnnotation(DBItemAttribute.class);
			if(_atr!=null){
				this.fieldMap.put(_atr.ColName(), f);
			}
			else{
				this.fieldMap.put(f.getName(), f);
			}
			
		}
		DBModelAttribute atr= cls.getAnnotation(DBModelAttribute.class);
    	if(atr!=null)
    	{
    		this.ItemCols=atr.ItemCols();
    		this.ListCols=atr.ListCols();
    		this.TBName=atr.TBName();
    	}
    	
//    	List<String> res=new ArrayList<String>();
//    	for(Field f:	cls.getDeclaredFields()){
//    	  
//    		SearchKeyAttribute _atr= f.getAnnotation(SearchKeyAttribute.class);
//    		if(atr!=null)
//    		{
//    			DBItemAttribute __atr=f.getAnnotation(DBItemAttribute.class);
//    			if(_atr!=null)
//    				res.add(__atr.ColName());
//    		}
//    	}
//    	this.SimpleCol= (String[]) res.toArray();
	}
	
	public EnitiyCache()
	{
	}

	public EnitiyCache(Class<?> cls) 
	{
		this.setCls(cls);
	}
}

File: Jazz.web.frame/net/Jazz.Common.Enitiy/EnitiyCache.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Reflection;

namespace Jazz.Common.Enitiy
{
    public  class EnitiyCache
    {
        private Type T;
	
	    private String TBName;
	
	    private String[] SimpleCol;
	
	    private String ItemCols;
	
	    private String ListCols;

        private Dictionary<String, PropertyInfo> fieldMap = new Dictionary<String, PropertyInfo>();

        public Dictionary<String, PropertyInfo> getfieldMap()
	    {
		    return fieldMap;
	    }

        public PropertyInfo getFieldName(String Colname)
	    {
		    return this.fieldMap[Colname];
	    }
	
	    public String getColName(String Fieldname)
	    {
		    foreach (var entry  in fieldMap) {
                if(entry.Value.Name==Fieldname)
                    return entry.Key;
		    }
		    return "";

	    }

	    public String getTBName() {
		    return TBName;
	    }

	    public String[] getSimpleCol() {
		    return SimpleCol;
	    }

	    public String getItemCols() {
		    return ItemCols;
	    }

	    public String getListCols() {
		    return ListCols;
	    }


	    public Type getCls() {
		    return T;
	    }

	
	    public void setCls<Enity>()  {
		    this.T=typeof(Enity);
		    this.fieldMap.Clear();
		    foreach(var f in T.GetProperties()){
			     IAttribute.DBItemAttribute _atr=f.GetCustomAttribute<IAttribute.DBItemAttribute>();
			    if(_atr!=null){
				    this.fieldMap.Add(_atr.ColName,f);
			    }
			    else{
				    this.fieldMap.Add(f.Name, f);
			    }
			
		    }
		    IAttribute. DBModelAttribute atr= T.GetCustomAttribute<IAttribute. DBModelAttribute>();
    	    if(atr!=null)
    	    {
    		    this.ItemCols=atr.ItemCols;
    		    this.ListCols=atr.ListCols;
    		    this.TBName=atr.TBName;
    	    }
    	
	    }
	
	    public EnitiyCache()
	    {
	    }
    }
}
